definitions/tools/merge_bams.cwl

```diff
@@ -3,26 +3,85 @@
 cwlVersion: v1.0
 class: CommandLineTool
 label: "Sambamba: merge"
-baseCommand: ["/usr/bin/sambamba_merge"]
+baseCommand: ["/bin/bash", "merge.sh"]
 requirements:
     - class: ResourceRequirement
       ramMin: 8000
       coresMin: 4
     - class: DockerRequirement
-      dockerPull: "mgibio/biscuit:0.3.8.2"
+      dockerPull: "mgibio/bam-merge:0.1"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'merge.sh'
+        entry: |
+            #!/bin/bash
+            set -o pipefail
+            set -o errexit
+            set -o nounset
+
+            SORTED=false
+
+            while getopts "t:m:n:s:" opt; do
+                case "$opt" in
+                    t)
+                        NTHREADS="$OPTARG"
+                        ;;
+                    m)
+                        MEM="$OPTARG"
+                        ;;
+                    n)
+                        OUTFILENAME="$OPTARG"
+                        ;;
+                    s)
+                        SORTED=true
+                        ;;
+                esac
+            done
+
+            BAMS=("${@:$OPTIND}")
+            NUM_BAMS=${#BAMS[@]}
+
+            #if there is only one bam, just copy it and index it
+            if [[ $NUM_BAMS -eq 1 ]]; then
+                cp "$BAMS" "$OUTFILENAME";
+            else
+                if [[ $SORTED == "true" ]];then
+                    /usr/bin/sambamba merge -t "$NTHREADS" "$OUTFILENAME" "$BAMS"
```

Comment:
I think `$BAMS` here needs to be updated to refer to the array.
```suggestion
                    /usr/bin/sambamba merge -t "$NTHREADS" "$OUTFILENAME" "${BAMS[@]}"
```
(It's fine in the `cp` case since it'll include the first entry, but that could also be updated to be an explicit reference to the first entry if desired.)

Reply:
Thank you for this review. I am listening.